storybookjs/storybook
# UI: Instant-on manager

Issue: #12702

## What I did

This implements an instant-on manager UI by bundling a prebuilt manager with the Storybook package, as well as caching a locally built manager. The cache gets busted whenever a change is made to `manager.js` or the webpack config (typically via `main.js`). There's also a `--no-manager-cache` flag which disables caching and ignores the prebuilt manager. The prebuilt manager is also used when building a static Storybook, assuming all criteria are met.

[▶️ Watch the demo](https://www.youtube.com/watch?v=lQLo4GpSPqw)

## How to test

- Run `start-storybook` against a vanilla, newly installed Storybook which only has `addon-links` and `addon-essentials` installed. It will say "Using prebuilt manager" and skip building the manager (see the startup message, it won't mention the manager build time).
- Run `build-storybook` against the same Storybook, and notice it will use the prebuilt manager here as well.
- Run `start-storybook` against any other Storybook. The first time it will build the manager as normal. Then kill and run `start-storybook` again. Now it will say "Using cached manager" and work the same as above.

The `--no-manager-cache` flag should be added to the docs.

## To do

- [x] Refactor the dev server for readability / maintainability
- [x] Start the browser as soon as the manager is ready
- [x] Show an appealing and useful UI as the preview builds
- [x] Use prebuilt when creating a static build
- [x] Setup our publish pipeline to prebuild the manager before publishing
